### PR TITLE
Add some tests for the Button widget

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets/button.tests.js [hunk 1]
```diff
@@ -113,6 +116,22 @@ QUnit.module('options changed callbacks', {
         assert.strictEqual(template.firstCall.args[0].custom, 1, 'custom field is correct');
     });
 
+    QUnit.test('stylingMode', function(assert) {
+        assert.ok(this.element.hasClass(BUTTON_CONTAINED_STYLE_CLASS));
+
+        this.instance.option('stylingMode', 'text');
+        assert.ok(this.element.hasClass(BUTTON_TEXT_STYLE_CLASS));
+        assert.notOk(this.element.hasClass(BUTTON_CONTAINED_STYLE_CLASS));
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/button.tests.js:126]: What about BUTTON_OUTLINED_STYLE_CLASS?

File: testing/tests/DevExpress.ui.widgets/button.tests.js [hunk 2]
```diff
@@ -113,6 +116,22 @@ QUnit.module('options changed callbacks', {
...
+
+        this.instance.option('stylingMode', 'outlined');
+        assert.ok(this.element.hasClass(BUTTON_OUTLINED_STYLE_CLASS));
+        assert.notOk(this.element.hasClass(BUTTON_TEXT_STYLE_CLASS));
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/button.tests.js:130]: What about BUTTON_CONTAINED_STYLE_CLASS?

File: testing/tests/DevExpress.ui.widgets/button.tests.js [hunk 3]
```diff
@@ -113,6 +116,22 @@ QUnit.module('options changed callbacks', {
...
+
+        this.instance.option('stylingMode', 'contained');
+        assert.ok(this.element.hasClass(BUTTON_CONTAINED_STYLE_CLASS));
+        assert.notOk(this.element.hasClass(BUTTON_OUTLINED_STYLE_CLASS));
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/button.tests.js:134]: What about BUTTON_TEXT_STYLE_CLASS?

File: testing/tests/DevExpress.ui.widgets/button.tests.js [hunk 4]
```diff
@@ -525,3 +544,39 @@ QUnit.module('templates', () => {
...
+        const params = clickHandler.getCall(0).args[0];
+        assert.ok(params, 'Event params should be passed');
+        assert.ok(params.event, 'Event should be passed');
+        assert.ok(params.validationGroup, 'validationGroup should be passed');
+    });
+
+    QUnit.test('contentReady', function(assert) {
+        assert.expect(3);
+
+        const eventHandler = (e) => {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/button.tests.js]: sinon.spy?

File: testing/tests/DevExpress.ui.widgets/button.tests.js (hunk 1 above)
Comment (alexander-kotov-dx) [reply testing/tests/DevExpress.ui.widgets/button.tests.js:126]: I check the widget has the new styling class and does not have the previous styling class. I think it is no reason to check other classes.

File: testing/tests/DevExpress.ui.widgets/button.tests.js (hunk 4 above)
Comment (alexander-kotov-dx) [reply testing/tests/DevExpress.ui.widgets/button.tests.js]: We can use sinon.spy here but i do not see any profits in this case

Comment (DokaRus) [reply testing/tests/DevExpress.ui.widgets/button.tests.js]: There is no need to extract handler. We can remove `const` and define handler as inline function.

```js
button.on('contentReady', (e) => {
 //handler
});
```

**Review by DokaRus — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets/button.tests.js (hunk 1 above)
Comment (San4es) [reply testing/tests/DevExpress.ui.widgets/button.tests.js:126]: ok

**Review by San4es — DISMISSED** [review]

**Review by DokaRus — DISMISSED** [review]

**Review by San4es — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Pull request merged**